app/models/help_text.rb

```diff
@@ -1,12 +1,21 @@
 class HelpText < ActiveRecord::Base
   attr_accessible :attribute_name, :instructions, :object_class, :title, :hint
-  validates_uniqueness_of :attribute_name, :scope => [:object_class]
+  validates_uniqueness_of :attribute_name, :scope => [:object_class, :audience]
   
   def self.for(object_class, attribute_name)
-    HelpText.for_object_class(object_class.to_s)[attribute_name.to_s]
+    HelpText.for_object_class(object_class.to_s)[attribute_name.to_s] 
+  end
+
+  def self.for(object_class, attribute_name, audience)
```

Comment:
You can probably merge this method with the above method.

Ruby has optional arguments, so you could make the method signature
```ruby
def self.for(object_class, attribute_name, audience = nil)
```
You could do the same for the method below
```ruby
def self.for_object_class(object_class, audience = nil)
```
and change the hash passed to `where` based on whether or not `audience is `nil`